libfreerdp/core/nla.c

```diff
@@ -1140,7 +1145,7 @@ SECURITY_STATUS nla_decrypt_public_key_echo(rdpNla* nla)
 		ap_integer_decrement_le(public_key2, public_key_length);
 	}
 
-	if (memcmp(public_key1, public_key2, public_key_length) != 0)
+	if (!public_key1 || !public_key2 || memcmp(public_key1, public_key2, public_key_length) != 0)
 	{
```

Comment:
if public_key1 or public_key2 are NULL that means that nla->packageName is not one of the one we know, so that means a real big bug that will have triggered some error long before. So these checks and initializations aren't needed.